ballerina/tests/mock_service.bal

```diff
@@ -0,0 +1,308 @@
+import ballerina/http;
+
+listener http:Listener httpListener = new (9090);
+
+service / on httpListener {
+
+    resource function get rest/api/'3/myself() returns User {
+        return {
+            "name": "Mia Krystof",
+            "active": true
+        };
+    }
+
+    // Mock response for `getMyPermissions`
+    resource function get rest/api/'3/mypermissions(http:Caller caller, http:Request req) returns error? {
```

Comment:
Better not to use caller. You can directly return from the resource function rather than using the `caller`. 

`resource function get rest/api/'3/mypermissions() returns Permissions|error? {`

Reply:
`http:Request req` is not being used anywhere. So you can omit that.

Reply:
Any reason for not using the Ballerina OpenAPI tool to generate the service? If you use that, it will generate the skeleton for you.
Ex:
    ```ballerina
resource function get rest/api/'3/mypermissions(string? projectKey, string? projectId, string? issueKey, string? issueId, string? permissions, string? projectUuid, string? projectConfigurationUuid, string? commentId) returns Permissions|ErrorCollectionBadRequest|ErrorCollectionUnauthorized|ErrorCollectionNotFound {
    }
```

Reply:
You can implement the function as follows,


```ballerina
resource function get rest/api/'3/mypermissions(string? projectKey, string? projectId, string? issueKey, string? issueId, string? permissions, string? projectUuid, string? projectConfigurationUuid, string? commentId) returns Permissions|ErrorCollectionBadRequest|ErrorCollectionUnauthorized|ErrorCollectionNotFound {
        return {
            permissions: {
                "BROWSE_PROJECTS": {
                    id: "10",
                    key: "BROWSE_PROJECTS",
                    name: "Browse Projects",
                    description: "Ability to browse projects and the issues within them.",
                    havePermission: true 
                }
            }
        };
}
```

Reply:
So I already had created that mock_service.bal before you told that skeleton method. I will make the necessary changes

Reply:
Applicable to all other resource methods in the service.

Reply:
@Bhashinee done, check mock_service.bal